app/routes/home.tsx

```diff
@@ -1,158 +1,215 @@
-                className={
-                  loginErrorResponse.fieldErrors?.password ? "error" : ""
-                }
-                type="password"
-                placeholder="Password"
-                onChange={handleLoginPassword}
-              />
-              <Button className="button-green" name="Login" type="submit" />
-              <div className="error-message">
-                {loginErrorResponse.fieldErrors?.name ??
-                  loginErrorResponse.fieldErrors?.password}
-              </div>
-            </form>
-          </div>
-
-          <div className="feedback-container">
-            <div className="caption">What do you think of Parrit?</div>
-            <a
-              className="text-link"
-              href="https://goo.gl/forms/ZGqUyZDEDSWqZVBP2"
-              target="_blank"
-              rel="noopener noreferrer"
-            >
-              Send feedback
-              <span className="carrot" />
-            </a>
-          </div>
-        </div>
-      </main>
-      <Footer />
-    </div>
-  );
+import homeStyles from "~/styles/home.css";
+
+export let links: LinksFunction = () => {
+    return [{ rel: "stylesheet", href: homeStyles }];
 };
 
-export default HomePage;
+export async function action({ request }: ActionFunctionArgs) {
+    let session = await getSession(request.headers.get("Cookie"));
+
+    let formData = await request.formData();
+    let _action = String(formData.get("_action"));
+    let email = String(formData.get("email"));
+    let password = String(formData.get("password"));
+
+    let supabaseClient = createSupabaseServerClient(request);
+
+    if (_action == "sign-up") {
+        let { accessToken, refreshToken, error } = await signUp(
+            email,
+            password,
+            supabaseClient
+        );
+
+        if (error) {
+            return { _action, error: error };
+        }
+        if (!accessToken || !refreshToken) {
+            return { _action, error: { message: "Something went wrong" } };
+        }
+        session = setAuthSession(session, accessToken, refreshToken);
+
+        return redirect("/project", {
+            headers: {
+                "Set-Cookie": await commitSession(session),
+            },
+        });
+    }
+
+    if (_action == "login") {
+        let { accessToken, refreshToken, error } = await signIn(
+            email,
+            password,
+            supabaseClient
+        );
+
+        if (error) {
+            return { _action, error: error };
+        }
+
+        if (!accessToken || !refreshToken) {
+            return {
+                _action,
+                error: { message: "Something is wrong with your credentials." },
+            };
+        }
+
+        session = setAuthSession(session, accessToken, refreshToken);
+
+        return redirect("/project", {
+            headers: { "Set-Cookie": await commitSession(session) },
+        });
+    }
+}
+
+export default function Home() {
```

Comment:
this edit, too, seems like it is more dramatic for having different spacing. We should look at the diff there before merging.

Reply:
I'm not sure if I addressed this comment with my spacing changes in the latest commit, let me know if not.